src/main/java/io/redlink/more/data/service/ElasticService.java

```diff
@@ -84,6 +89,41 @@ public List<String> storeDataPoints(final List<DataPoint> dataBulk, final Routin
         }
     }
 
+
+    public long deleteDataPoints(RoutingInfo routingInfo, String key, Set<String> filteredByValues) throws IOException {
+        if (filteredByValues.isEmpty() || key == null) {
+            return 0L;
+        }
+        final String indexName = "study_" + routingInfo.studyId();
+        DeleteByQueryRequest request = new DeleteByQueryRequest.Builder()
+                .index(indexName)
+                .query(q -> q
+                        .bool(b -> b
+                                .must(m -> m.term(t -> t.field("study_id")
+                                        .value("study_" + routingInfo.studyId())))
+                                .must(m -> m.term(t -> t.field("participant_id")
+                                        .value("participant_" + routingInfo.participantId())))
+                                .must(m -> m.bool(bb -> bb
+                                        .should(s -> s.terms(t -> t
+                                                .field(key)
+                                                .terms(v -> v.value(
+                                                        filteredByValues.stream()
+                                                                .map(FieldValue::of)
+                                                                .toList()
+                                                ))
+                                        ))
+                                        .minimumShouldMatch("1")
+                                ))
+                        )
+                )
```

Comment:
Ich schätze mal, dass das so funktionieren sollte:

```suggestion
                .query(q -> q
                        .bool(b -> b
                                .filter(studyFilter -> studyFilter
                                        .term(t -> t.field("study_id")
                                                .value("study_" + routingInfo.studyId())))
                                .filter(participantFilter -> participantFilter
                                        .term(t -> t.field("study_id")
                                                .value("study_" + routingInfo.studyId())))
                                .filter(summaryIdFilter -> summaryIdFilter
                                        .terms( terms -> terms.field(key)
                                                .terms(values -> values.value(filteredByValues.stream()
                                                        .map(FieldValue::of)
                                                        .toList()))))
                        )
                )
                .build();
```

Reply:
https://stackoverflow.com/questions/72197344/creating-a-termquery-with-a-list-by-using-elasticsearch-java-api-client